jib-core/src/main/java/com/google/cloud/tools/jib/frontend/BuildStepsRunner.java

```diff
@@ -255,6 +257,14 @@ public void build(HelpfulSuggestions helpfulSuggestions) throws BuildStepsExecut
         throw new BuildStepsExecutionException(
             helpfulSuggestions.forInsecureRegistry(), exceptionDuringBuildSteps);
 
+      } else if (exceptionDuringBuildSteps instanceof RegistryErrorException) {
+        // RegistryErrorExceptions have good messages
+        RegistryErrorException registryErrorException =
+            (RegistryErrorException) exceptionDuringBuildSteps;
+        String message =
+            Verify.verifyNotNull(registryErrorException.getMessage()); // keep null-away happy
```

Comment:
Perhaps we should make `RegistryErrorException#getMessage` not be `nullable`?

Reply:
I tried, providing an override on `RegistryErrorException` for `getMessage()` with both `javax.annotation.Nonnull` (from jsr305) and Checker's `org.checkerframework.checker.nullness.qual.NonNull`, but NullAway refuses to cooperate.